Tests/SwiftProtobufTests/Test_Enum.swift

```diff
@@ -40,6 +41,24 @@ class Test_Enum: XCTestCase, PBTestHelpers {
         }
     }
 
+    func testJSONdecodingOptions() {
+        var options = JSONDecodingOptions()
+        options.ignoreUnknownFields = true
+
+        assertJSONDecodeSucceeds("{\"optionalNestedEnum\":\"NEW_VALUE\"}", options: options) { (m: MessageTestType) -> Bool in
+            switch m.optionalNestedEnum {
+            case .zero: return true
+            default: return false
+            }
+        }
+
+        assertJSONDecodeSucceeds("{\"repeatedNestedEnum\":[\"FOO\",\"NEW_VALUE\"]}", options: options) { (m: MessageTestType) -> Bool in
+            m.repeatedNestedEnum == [.foo]
+        }
+
+        // Proto3Unittest_TestAllTypes doesn't have a map<>, Test_Map_JSON covers this case.
+    }
+
```

Comment:
Do we need to add a test case to verify that we reject unknown enums without this option?

Reply:
Was hoping we had that, but added locally to make it explicit.